cpp/examples/arrow/row_wise_conversion_example.cc

```diff
@@ -61,51 +65,52 @@ arrow::Status VectorToColumnarTable(const std::vector<struct data_row>& rows,
   arrow::MemoryPool* pool = arrow::default_memory_pool();
 
   Int64Builder id_builder(pool);
-  DoubleBuilder cost_builder(pool);
-  ListBuilder components_builder(pool, std::make_shared<DoubleBuilder>(pool));
-  // The following builder is owned by components_builder.
-  DoubleBuilder& cost_components_builder =
-      *(static_cast<DoubleBuilder*>(components_builder.value_builder()));
+  Int64Builder components_builder(pool);
+  ListBuilder component_cost_builder(pool, std::make_shared<DoubleBuilder>(pool));
+  // The following builder is owned by component_cost_builder.
+  DoubleBuilder& component_item_cost_builder =
+      *(static_cast<DoubleBuilder*>(component_cost_builder.value_builder()));
```

Comment:
I think it would be easier to follow if we made `component_item_cost_builder` a pointer instead of a reference.
That would make much more obvious that `component_item_cost_builder` is the `value_builder` of `component_cost_builder` and we are not creating anything new.

Reply:
Done

Reply:
@amol- Done